src/typegraph/core/src/global_store.rs

```diff
@@ -157,9 +158,8 @@ impl Store {
         with_store(|s| s.type_by_names.get(name).map(|id| id.0.clone()))
     }
 
-    pub fn register_type_ref(builder: TypeRefBuilder) -> Result<TypeRef> {
-        let id = with_store(|s| s.types.len()) as u32;
-        let type_ref = builder.with_id(id.into());
+    pub fn assign_type_ref(id: TypeId, builder: TypeRefBuilder) -> Result<TypeRef> {
+        let type_ref = builder.with_id(id);
```

Comment:
This looks similar in spirit to [this](https://github.com/metatypedev/metatype/blob/main/src/typegraph/core/src/runtimes/substantial/type_utils.rs#L64) now that I think about it, mindlessly simple but working.

Reply:
Good point. The current code is not using the replace functionality, only setting empty slots. Have it so that it panics if the slot is already filled?

Reply:
> Have it so that it panics if the slot is already filled?

That would work. But I am not really sure why you'd be required to play with the ids directly, either way if an id is allocated that means a named type is there already so why not just use the names directly... So why not just use a  t::ref(name)?

Reply:
Oooh, that just might work. Let me give it a try...

Reply:
Oh yeah, I don't know how I missed a ref based solution. Reduces my diff to a few lines! Thanks.